docs/examples/example_onnx.ipynb

```diff
@@ -0,0 +1,349 @@
+    "```\n",
+    "\n",
+    "### Converting the base models to ONNX\n",
+    "\n",
+    "Before being able to convert the MetaLearner to ONXX we need to manually convert the necessary\n",
+    "base models for the prediction. To get the necessary base models that need to be\n",
+    "converted we can use {meth}`~metalearners.metalearner.MetaLearner._necessary_onnx_models`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "necessary_models = xlearner._necessary_onnx_models()\n",
+    "necessary_models"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We see that we need to convert the ``\"propensity_model\"``, the ``\"control_effect_model\"``\n",
+    "and the ``\"treatment_effect_model\"``. We can do this with the following code where we\n",
+    "use the ``convert_lightgbm`` function from the ``onnxmltools`` package.\n",
+    "\n",
+    "```{note}\n",
+    "It is important to know that for classifiers we need to pass the ``zipmap=False`` option. This\n",
+    "is required so the output probabilities are a Matrix and not a list of dictionaries.\n",
+    "In the case of using a ``sklearn`` model and using the ``convert_sklearn`` function, this\n",
+    "option needs to be specified with the ``options={\"zipmap\": False}`` parameter.\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import onnx\n",
+    "from onnxmltools import convert_lightgbm\n",
+    "from onnxconverter_common.data_types import FloatTensorType\n",
+    "\n",
+    "onnx_models: dict[str, list[onnx.ModelProto]] = {}\n",
+    "\n",
+    "for model_kind, models in necessary_models.items():\n",
+    "    onnx_models[model_kind] = []\n",
+    "    for model in models:\n",
+    "        onnx_models[model_kind].append(\n",
+    "            convert_lightgbm(\n",
+    "                model,\n",
+    "                initial_types=[(\"X\", FloatTensorType([None, len(feature_columns)]))],\n",
+    "                zipmap=False,\n",
+    "            )\n",
+    "        )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we can call {meth}`~metalearners.metalearner.MetaLearner._build_onnx` which combines\n",
+    "the the converted ONNX base models into a single ONNX model.\n",
+    "This combined model has a single 2D input ``\"X\"`` and a single output named ``\"tau\"``.\n",
+    "The output name can be changed using the ``output_name`` parameter. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "onnx_model = xlearner._build_onnx(onnx_models)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can explore the input and output of the model and see that it expects a matrix with 11\n",
```

Comment:
You might want to add a visualization of the model using [netron.app](https://netron.app/).

Reply:
I added a visualization:
https://metalearners--39.org.readthedocs.build/en/39/examples/example_onnx.html

Reply:
![image](https://github.com/user-attachments/assets/7d5bca0c-ef3d-413a-bb0e-abace062bd23)

This looks a bit odd. Do you understand what is going on here? Why the slicing and the cast that leads nowhere? Is that something skl2onnx did?

Reply:
I'm afraid I wound't know :/